Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -163,18 +218,35 @@ private class SymbolDeclarationResolver {
           cancellationToken.ThrowIfCancellationRequested();
           methodSymbol.Returns.Add(ProcessFormal(scope, result));
         }
-        ProcessAndRegisterMethodBody(methodSymbol, method.Body);
+        methodSymbol.Block = ProcessMethodBody(methodSymbol, method.Body);
+        Func<Expression, ScopeSymbol?> subP = e => ProcessMethodExpr(methodSymbol, e);
+        methodSymbol.Ens = ProcessListAttributedExpressions(method.Ens, subP);
+        methodSymbol.Req = ProcessListAttributedExpressions(method.Req, subP);
+        methodSymbol.Mod = ProcessListExpressions(method.Mod.Expressions.Select(e => e.E).ToList(), subP);
+        methodSymbol.Decreases = ProcessListExpressions(method.Decreases.Expressions, subP);
```

Comment:
These properties might be unnecessary for this PR.

Reply:
As said in a previous comment of this same thread, they are useful and even necessary now that we introduce expression-based scopes.